import-automation/executor/app/service/github_api.py

```diff
@@ -0,0 +1,374 @@
+        })
+        response = requests.get(content_query, auth=self.auth)
+        response.raise_for_status()
+        content_info = response.json()
+        return [
+            entry['name'] for entry in content_info if entry['type'] == 'file'
+        ]
+
+    def _dir_exists(self, commit_sha: str, dir_path: str) -> bool:
+        """Checks whether a directory still exists in the repository
+        after a commit.
+
+        Args:
+            commit_sha: Commit ID as a string
+            dir_path: Path to the directory to check as a string.
+
+        Returns:
+            True if the directory exists. False, otherwise.
+        """
+        content_query = self._build_content_query(commit_sha, dir_path)
+        response = requests.get(content_query, auth=self.auth)
+        is_ok = response.status_code == http.HTTPStatus.OK
+        is_not_found = response.status_code == http.HTTPStatus.NOT_FOUND
+        if not is_ok and not is_not_found:
+            response.raise_for_status()
+        return response.status_code != http.HTTPStatus.NOT_FOUND
+
+    def _path_containing_file(self, commit_sha: str, dir_path: str,
+                              containing: str) -> bool:
+        """Checks if the directory in the repository contains a file.
+
+        Example:
+            There is a README.md in foo/bar when commit '21d12fr2' happens.
+            The method call
+                _path_containing_file('21d12fr2', 'foo/bar', 'README.md')
+            returns True.
+
+        Args:
+            commit_sha: Commit ID as a string.
+            dir_path: Path of the directory to search in as a string.
+            containing: Name of the file to search for as a string.
+
+        Returns:
+            True if the directory contains a file with the given name. False,
+            otherwise.
+        """
+        files_in_dir = self._query_files_in_dir(commit_sha, dir_path)
+        for filename in files_in_dir:
+            if filename == containing:
+                return True
+        return False
+
+    def _find_dirs_up_path_containing_file(self, commit_sha: str, dir_path: str,
+                                           containing: str,
+                                           searched: Set[str]) -> Set[str]:
+        """Given a path pointing at a directory in the repository, searches for
+        a file in the directory and its parent directories.
+
+        Example:
+            Assume the repository has a subdirectory called 'scripts' and
+            'scripts' has a subdirectory called 'us_fed'.
+            There is a 'README.md' in the root directory of the repository and
+            another in 'us_fed'.
+
+            The method call
+                _find_dirs_up_path_containing_file(
+                    'dw1f231asdf1321', scripts/us_fed', 'README.md', set())
+            first searches in 'us_fed' and adds 'scripts/us_fed' to
+            'found_dirs'. It then searches in 'scripts' and the root directory
```

Comment:
This seems like order in the returned value is important.  Is this true?

Reply:
The method returns a set, so there isn't really any order